test/test_autograd.py

```diff
@@ -2831,21 +2831,23 @@ def check(name):
                 args_variable, kwargs_variable = create_input(args, requires_grad=not is_inplace, call_kwargs=kwargs)
                 self_tensor = deepcopy(self_variable.data)
                 args_tensor = deepcopy(unpack_variables(args_variable))
-                output_variable = getattr(self_variable, name)(*args_variable, **kwargs_variable)
-                if not exclude_tensor_method(name, test_name):
-                    output_tensor = getattr(self_tensor, name)(*args_tensor, **kwargs_variable)
-                    if not isinstance(output_tensor, torch.Tensor) and not isinstance(output_tensor, tuple):
-                        output_tensor = torch.DoubleTensor((output_tensor,))
-                    self.assertEqual(unpack_variables(output_variable), output_tensor)
-                    # TODO: check that both have changed after adding all inplace ops
-
-                def fn(*inputs):
-                    output = getattr(inputs[0], name)(*inputs[1:], **kwargs)
-                    return output_process_fn(output)
-
-                if not is_inplace and name not in EXCLUDE_GRADCHECK:
-                    run_grad_and_gradgrad_checks(self, name, test_name, fn,
-                                                 output_variable, (self_variable,) + args_variable)
+                is_method = hasattr(self_variable, name)
+                if is_method:
+                    output_variable = getattr(self_variable, name)(*args_variable, **kwargs_variable)
+                    if not exclude_tensor_method(name, test_name):
```

Comment:
is exclude_tensor_method here not sufficient?

Reply:
no, it is not sufficient, the line above it (2834) will fail if the name is not a tensor method.

Reply:
2836 in the new code

Reply:
FYI you can link the code directly, click "view file" under "Files Changed" such as 
`https://github.com/pytorch/pytorch/blob/c85170bd2d0bc09c4bc91c621b96345b01069f16/test/test_autograd.py#L2836`

which is displayed as:
https://github.com/pytorch/pytorch/blob/c85170bd2d0bc09c4bc91c621b96345b01069f16/test/test_autograd.py#L2836.